### PR TITLE
New: Documentation ToC layout on mobile

### LINKED ISSUE #198: Design new solution for tree view/TOC on mobile
Currently it appears before the content but the tree view can be incredibly long. If it appears after the content, there are a11y concerns about it being used as navigation since the user would have to go through all the content to get to the bottom. 

Will look at different options, either a dropdown or a flyout menu on mobile.

### PR DESCRIPTION
**Pull request by ststimac**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/contributing/pull-requests/#commitmessages)

## Short description of the change(s)

Front-end for dropdown table of contents for the documentation pages on mobile/smaller screens.

Fix #198

**I need help getting the dropdown to open**, and placeholders for what links that are currently in place as I assume there's some handlebars code to be written.

Since this is using a different structure from the table of contents on the larger screens, I don't know what can be reused from the handlebars code.

### CODE REVIEW

**Comment by qzhou1607** [comment]
The dropdown content and javascript was added.

**Comment by ststimac** [comment]
Thank you @qzhou1607

File: src/hexo/themes/sonarwhal/layout/partials/sub-page.hbs [hunk 1]
```diff
@@ -30,10 +30,35 @@
...
     <div class="section">
         <div class="layout layout--sidebar">
-            <div class="module module--primary wrap-text">
-                {{{page.content}}}
+            <div class="module module--secondary mobile-view">
+                <ul class="toc-mobile-view">
+                    <li>
+                        <p role="button" id="mobile-toc" aria-controls="mobile-toc-submenu" aria-expanded="false" aria-haspopup="true" class="toc-mobile-view__button expandable">{{capitalize page.category}}</p>
+                        <ul role="group" class="toc-mobile-view__dropdown" id="mobile-toc-submenu" aria-hidden="true">
+                             {{#if (isNotGuideIndexPage page)}}
```
Comment (alrra) [inline src/hexo/themes/sonarwhal/layout/partials/sub-page.hbs]: Indent is a bit off here.

**Comment by ststimac** [comment]
I'm only seeing this on the User Guide. Is there another template for the Contributor Guide and Rules?

The contributor guide and rules should have the same layout.

**Comment by molant** [comment]
Maybe the service worker?

**Comment by ststimac** [comment]
@molant but it *is* showing on the user guide. just not contributor guide or rules.

**Comment by molant** [comment]
Rules:
image

User guide:
image

Contributor guide:
image


Screenshots from private mode. Does anything not look the way it should be?

**Comment by ststimac** [comment]
Must have been the service worker then :S

**Comment by qzhou1607** [comment]
@ststimac Sometimes the updates don't show when the old service worker is still in control even though there is updated content in cache. Close all tabs and reopen (not refresh) helps.

**Pull request closed without merging**